Collider.h for Entities, Collision.cpp/Colliion.h for handling collis…

cmg7063 opened the pull request:
Are GameEntities == GameObjects (collectibles, obstacles, etc)? If not, I recommend creating a GameObject class which takes a Mesh, Material, and Collider type

ttanatb commented:
GameEntities are pretty much GameObjects. They already have a pointer to a mesh, and material, so feel free to add a pointer to the collider class in the GameEntities class

ttanatb commented:
Not sure if something's supposed to happen, but you forgot to load the cube mesh and the skybox didn't like that.

cmg7063 commented:
I added imgui to the project and used it in the CollisionTestScene so you can evoke collisions between entities. I added collider params to the mesh so MeshManager creates a mesh with a specified collider (there can still be meshes without colliders). I tried making the collider member in GameEntity a pointer but when setting the pointer to &Collider in the parameterized GameEntity objs it wasn’t handling collisions. I'm not that well versed with pointers so it's probably a mistake on my part with implementing it.
